frb_codegen/src/method_utils.rs

```diff
@@ -0,0 +1,181 @@
+use crate::ir::IrType::{self, StructRef};
+use crate::ir::{IrField, IrTypeStructRef};
+use crate::ir::{IrFile, IrFunc};
+use convert_case::{Case, Casing};
+
+const STATIC_METHOD_MARKER: &str = "__static_method___";
+const METHOD_MARKER: &str = "__method___";
+
+pub(crate) struct MethodNamingUtil;
+
+impl MethodNamingUtil {
+    // Is the function name for a non static method?
+    pub fn is_non_static_method(s: &str) -> bool {
+        s.contains(METHOD_MARKER)
+    }
+
+    // Is the function name for a static method?
+    pub fn is_static_method(s: &str) -> bool {
+        s.contains(STATIC_METHOD_MARKER)
+    }
+
+    // Tests if the function is `f` is a static method for struct with name `struct_name`
+    pub fn is_static_method_for_struct(f: &&IrFunc, struct_name: &str) -> bool {
+        f.name.contains(STATIC_METHOD_MARKER)
+            && f.name.split(STATIC_METHOD_MARKER).last().unwrap() == struct_name
+    }
+
+    //Does `ir_file` has any methods directed for `struct_name`?
+    pub fn has_methods(struct_name: &String, ir_file: &IrFile) -> bool {
+        ir_file.funcs.iter().any(|f| {
+            Self::is_method_for_struct(&f, struct_name)
```

Comment:
About the `&&IrFunc` problem - can we directly pass `f` not `&f` here? (And all similar places). Since I guess `&IrFunc` is `Copy`?